enzyme/Enzyme/BlasDerivatives.td

```diff
@@ -117,81 +147,103 @@ def copy : CallBlasPattern<(Op $n, $x, $incx, $y, $incy),
 //                   ]
 //                   >;
 
-def asum : CallBlasPattern<(Op $n, $x, $incx),
-                  [],[len, vinc],
-                  [(noop)]// sum(abs(x))
-                  >;
-
+// def asum : CallBlasPattern<(Op $n, $x, $incx),
+//                   [],[len, vinc],
+//                   [(b<"axpy"> DiffeRet<"">, sign(<input<"x">), $y)]
+//                   >;
+// 
 
 // Lv 2
 
 // y = alpha Ax + beta y
-// def gemv : CallBlasPattern<(Op $layout, $transa, $m, $n, $alpha, $A, $lda, $x, $incx, $beta, $y, $incy),
-//                   [cblas_layout, trans, len, len, fp, mld, vinc, fp, vinc],
-//                   [
-//                   /* alpha  */  (b<"dot"> DiffeRet<"">, (b<"gemv"> $A, $x)),
-//                   /* A      */  //(),
-//                   /* x      */  //(),
-//                   /* beta   */  (b<"dot"> $y, DiffeRet<"">),
-//                   /* y      */  (b<"scal"> $beta, DiffeRet<"">),
-//                   ]
-//                   >;
+// currently assumes for vector dimensions that transa = 'N' and gets dimensions wrong otherwise
```

Comment:
Can we check that `transa` is `N`? Is this comment still valid?

Reply:
I did write one xFail test where trans is constInst T instead of N.
I just try to cover the second case with a test  where trans isn't a compile time constant.